purchases/src/androidTestDefaults/kotlin/com/revenuecat/purchases/PurchasesIntegrationTest.kt

```diff
@@ -205,6 +209,172 @@ class PurchasesIntegrationTest : BasePurchasesIntegrationTest() {
+                        validateAllZeroBalances(virtualCurrencies = virtualCurrencies)
+                        lock.countDown()
+                    },
+                )
+            },
+        )
+
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    @Test
+    fun testCachedVirtualCurrencies() {
+        val appUserID = "integrationTestUserWithAllBalancesNonZero"
+        val lock = CountDownLatch(1)
+
+        Purchases.sharedInstance.logInWith(
+            appUserID = appUserID,
+            onError = { error -> fail("should have been able to login. Error: $error") },
+            onSuccess = { _, created ->
+                assertThat(created).isFalse() // This user should be already exist
+
+                Purchases.sharedInstance.invalidateVirtualCurrenciesCache()
+
+                Purchases.sharedInstance.getVirtualCurrenciesWith(
+                    onError = { error -> fail("should be success. Error: $error") },
+                    onSuccess = { virtualCurrencies ->
+                        validateNonZeroBalances(virtualCurrencies = virtualCurrencies)
+
+                        var cachedVirtualCurrencies = Purchases.sharedInstance.cachedVirtualCurrencies
+                        validateNonZeroBalances(virtualCurrencies = cachedVirtualCurrencies)
+
+                        Purchases.sharedInstance.invalidateVirtualCurrenciesCache()
+                        cachedVirtualCurrencies = Purchases.sharedInstance.cachedVirtualCurrencies
+                        assertThat(cachedVirtualCurrencies).isNull()
+
+                        lock.countDown()
+                    },
+                )
+            },
+        )
+
+        lock.await(testTimeout.inWholeSeconds, TimeUnit.SECONDS)
+        assertThat(lock.count).isZero
+    }
+
+    private fun validateAllZeroBalances(virtualCurrencies: VirtualCurrencies?) {
+        validateVirtualCurrenciesObject(
+            virtualCurrencies = virtualCurrencies,
+            testVCBalance = 0,
+            testVC2Balance = 0,
+        )
+    }
+
+    private fun validateNonZeroBalances(virtualCurrencies: VirtualCurrencies?) {
+        validateVirtualCurrenciesObject(
+            virtualCurrencies = virtualCurrencies,
+            testVCBalance = 100,
+            testVC2Balance = 777,
+        )
+    }
+
+    @Suppress("MagicNumber")
+    private fun validateVirtualCurrenciesObject(
+        virtualCurrencies: VirtualCurrencies?,
+        testVCBalance: Int,
+        testVC2Balance: Int,
+        testVC3Balance: Int = 0,
+    ) {
+        assert(virtualCurrencies!!.all.count() == 3)
+
+        val expectedTestVirtualCurrency = VirtualCurrency(
+            code = "TEST",
+            name = "Test Currency",
+            balance = testVCBalance,
+            serverDescription = "This is a test currency",
+        )
+        assert(virtualCurrencies["TEST"] == expectedTestVirtualCurrency)
```

Comment:
Another nitpick. I believe this works! But we've been using `AssertJ` assertions for these since they are a bit more readable I think... NABD though!
```suggestion
        assertThat(virtualCurrencies["TEST"]).isEqualTo(expectedTestVirtualCurrency)
```